src/base/random.hh

```diff
@@ -99,18 +99,22 @@ class Random
     random()
     {
         // [0, 1) for real types
-        std::uniform_real_distribution<T> dist;
-        return dist(gen);
+        warn("FP random numbers are not uniformly distributed.");
+        return (T) random<uint64_t>() /
+          (T) std::numeric_limits<uint64_t>::max();
     }
+
     /**
      * @ingroup api_base_utils
      */
     template <typename T>
     typename std::enable_if_t<std::is_integral_v<T>, T>
     random(T min, T max)
     {
-        std::uniform_int_distribution<T> dist(min, max);
-        return dist(gen);
+        assert(min <= max);
+        // + 1 to handle cases where min == max
+        T r = (gen() - min) % (max - min + 1) + min;
```

Comment:
Why `- min`? I think this should be:
        T r = gen() % (max - min + 1) + min;

Reply:
Yes, that works too. Probably I tried to balance it (-min then +min) but it does not make sense since we are working with unsigned.